qp/utils.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+import sys
+
+def safelog(arr, threshold=sys.float_info.epsilon):
+    """
+    Takes log of array with zeroes.
+
+    Parameters
+    ----------
+    arr: ndarray
+        Values to be logged
+    threshold: float
+        Small, positive value to replace zeros and negative numbers
+
+    Returns
+    -------
+    logged: ndarray
+        Logged values, with approximation in place of zeros and negative numbers
+    """
+    shape = np.shape(arr)
+    flat = arr.flatten()
+    logged = np.log(np.array([max(a,threshold) for a in flat])).reshape(shape)
+    return logged
+
+def calckl(qn, pn):
+    """
+    Calculates Kullback-Leibler Divergence
+
+    Parameters
+    ----------
+    qn: ndarray
```

Comment:
I'm removing the PDF.kld function and only using the qp.utils.calculate_kl_divergence function, which now uses evaluate() instead of specifying the mode of evaluation.  I added a test case into the notebook for a PDF object initialized with quantiles and then calculated the KLD from the quantile-initialized PDF object to the truth-initialized PDF object.  Does that look good to you?